frontend/src/pages/SignUp/index.tsx

```diff
@@ -98,69 +72,94 @@ const SignUp: React.FC = () => {
-              emptyField={emptyFields.firstName}
-              validator={nameValidator}
-              onChange={(value, isValid) =>
-                handleInputChange("firstName", value, isValid)
-              }
+              fullWidth
+              margin="normal"
+              {...register("firstName", { validate: { nameValidator } })}
+              error={!!errors.firstName}
+              helperText={errors.firstName?.message}
             />
-            <CustomTextField
-              label="Last Name"
-              size="small"
+            <TextField
+              label="Last name"
               required
-              emptyField={emptyFields.lastName}
-              validator={nameValidator}
-              onChange={(value, isValid) =>
-                handleInputChange("lastName", value, isValid)
-              }
+              fullWidth
+              margin="normal"
+              {...register("lastName", { validate: { nameValidator } })}
+              error={!!errors.lastName}
+              helperText={errors.lastName?.message}
             />
-            <CustomTextField
+            <TextField
               label="Username"
-              size="small"
               required
-              emptyField={emptyFields.username}
-              validator={usernameValidator}
-              onChange={(value, isValid) =>
-                handleInputChange("username", value, isValid)
-              }
+              fullWidth
+              margin="normal"
+              {...register("username", { validate: { usernameValidator } })}
+              error={!!errors.username}
+              helperText={errors.username?.message}
             />
-            <CustomTextField
+            <TextField
               label="Email"
-              size="small"
               required
-              emptyField={emptyFields.email}
-              validator={emailValidator}
-              onChange={(value, isValid) =>
-                handleInputChange("email", value, isValid)
-              }
+              fullWidth
+              margin="normal"
+              type="email"
+              {...register("email", { validate: { emailValidator } })}
+              error={!!errors.email}
+              helperText={errors.email?.message}
             />
-            <CustomTextField
+            <TextField
               label="Password"
-              size="small"
               required
-              emptyField={emptyFields.password}
-              validator={passwordValidator}
-              onChange={(value, isValid) =>
-                handleInputChange("password", value, isValid)
-              }
-              isPasswordField
+              fullWidth
+              margin="normal"
+              type="password"
+              {...register("password", { validate: { passwordValidator } })}
+              error={!!errors.password}
+              helperText={errors.password?.message}
+              typeof={showPassword ? "text" : "password"}
```

Comment:
Should it be `type={showPassword ? "text" : "password"}`

And then I think line 131 can be removed!